bertopic/_bertopic.py

```diff
@@ -860,42 +817,54 @@ def _extract_topics(self, documents: pd.DataFrame):
         self._extract_words_per_topic(words)
         # self._create_topic_vectors()  # we don't use this feature (topic search with keyword)
 
-    def _create_topic_vectors(self):
+    def _create_topic_vectors(self, embedding_model=None, method='doc',
+                              embeddings=None, predictions=None):
         """ Creates embeddings per topics based on their topic representation
-
         We start by creating embeddings out of the topic representation. This
         results in a number of embeddings per topic. Then, we take the weighted
         average of embeddings in a topic by their c-TF-IDF score. This will put
         more emphasis to words that represent a topic best.
-
         Only allow topic vectors to be created if there are no custom embeddings and therefore
         a sentence-transformer model to be used or there are custom embeddings but it is allowed
         to use a different multi-lingual sentence-transformer model
         """
-        if not self.custom_embeddings or all([self.custom_embeddings and self.allow_st_model]):
-            topic_list = list(self.topics.keys())
-            topic_list.sort()
-            n = self.top_n_words
+        topic_list = list(filter(lambda x: x != -1, list(self.topics.keys())))
+        topic_list.sort()  # todo: might be unncessary, but from original bertopic
+        topics_embeddings = list()
+        if method == 'word':
+            if not embedding_model:
+                raise ValueError("Need embedding model for \"word\" topic vectors.")
+            n = len(self.get_topic(topic_list[0]))  # just to get how many
+            # words represents a topic
 
             # Extract embeddings for all words in all topics
             topic_words = [self.get_topic(topic) for topic in topic_list]
             topic_words = [word[0] for topic in topic_words for word in topic]
-            embeddings = self._extract_embeddings(topic_words)
+            embeddings = self._extract_embeddings(topic_words, embedding_model=embedding_model)
 
-            # Take the weighted average of word embeddings in a topic based on their c-TF-IDF value
+            # Take the weighted average of word embeddings in a
+            # topic based on their c-TF-IDF value
             # The embeddings var is a single numpy matrix and therefore slicing is necessary to
             # access the words per topic
-            topic_embeddings = []
             for i, topic in enumerate(topic_list):
                 word_importance = [val[1] for val in self.get_topic(topic)]
                 if sum(word_importance) == 0:
                     word_importance = [1 for _ in range(len(self.get_topic(topic)))]
                 topic_embedding = np.average(embeddings[i * n: n + (i * n)],
                                              weights=word_importance,
                                              axis=0)
```

Comment:
Add a note here about potential issues that may occur in word index-based topic vector extraction.